### PR TITLE
Issue/48 dependency handling

### PR DESCRIPTION
**Pull request by wouterdb**

### CODE REVIEW

File: plugins/__init__.py [hunk 1]
```diff
@@ -642,113 +644,97 @@ class EndPoint(KeystoneResource):
...
-
-        elif res.id.entity_type == "openstack::SecurityGroup":
-            if not res.purged:
-                sgs[res.name] = res
-
     # they require the tenant to exist
     for network in networks.values():
         if network.model.project.name in projects:
-            network.requires.add(projects[network.model.project.name])
+            network.requires.add(projects[network.model.project.name].id)
```
Comment (bartv) [inline plugins/__init__.py]: I think both are possible: the resource or the id

Comment (wouterdb) [reply plugins/__init__.py]: it is, but testing is a lot easier if the type is uniform

File: tests/test_dependency_handling.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    assert_before("VirtualMachine", "Project")
+    assert_before("VirtualMachine", "Subnet")
+
+    assert_before("HostPort", "Project")
+    assert_before("HostPort", "VirtualMachine")
+
+    assert_before("FloatingIP", "Router")
+    assert_before("FloatingIP", "HostPort")
+    assert_before("FloatingIP", "Network")
```
Comment (arnaudsjs) [inline tests/test_dependency_handling.py]: SecurityGroup is not covered.

File: tests/test_dependency_handling.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            continue
+        agentname = name.get_agent_name()
+        for r in resource.requires: 
+            #ensure clean typing
+            assert isinstance(r, Id)
+            if not "openstack" in r.get_entity_type():
+                continue
+            assert agentname == r.get_agent_name()
+
+    def assert_before(typea, typeb, tenant="t1"):
```
Comment (arnaudsjs) [inline tests/test_dependency_handling.py]: This name confused me. I was expecting that typea would be executed before typeb, but it's to other way around.

**Pull request merged**